ignite/distributed/comp_models/horovod.py

```diff
@@ -165,13 +164,18 @@ def spawn(  # type: ignore[override]
 
         _manual_reduce_op_map = {"MIN": torch.min, "MAX": torch.max, "PRODUCT": torch.prod}
 
-        def _do_all_reduce(self, tensor: torch.Tensor, op: str = "SUM") -> torch.Tensor:
+        def _do_all_reduce(self, tensor: torch.Tensor, op: str = "SUM", group: Optional[Any] = None) -> torch.Tensor:
             if op in self._manual_reduce_op_map:
                 op_fn = self._manual_reduce_op_map[op]
                 return self._do_manual_all_reduce(tensor, op_fn)
             if op not in self._reduce_op_map:
                 raise ValueError(f"Unsupported reduction operation: '{op}'")
             op = self._reduce_op_map[op]
+            if group is not None:
+                if not isinstance(group, hvd.ProcessSet):
+                    raise ValueError("Argument group should be list of int or ProcessSet")
+                hvd.init(process_sets=[group])
```

Comment:
@puhuk do you know what `hvd.init(process_sets=[group])` does ?

Reply:
and how it works if horovod is already initialized ?

Reply:
It sets up the horovod backend again by calling `_setup_process_sets`

Reply:
you have to check that

Reply:
There are two options in initializing horovod
1. Static initialization like `hvd.init(process_sets= ~~)`   --> In this case, when initialized, process_set can not be changed
2. Dynamic process set like `hvd.init(process_sets="dynamic") --> In this case, process set can be altered dynamically